services/ui-src/src/utils/auth/userProvider.test.tsx

```diff
@@ -107,37 +99,82 @@ describe("Test UserProvider", () => {
     expect(screen.getByTestId("testdiv")).toHaveTextContent("User Test");
   });
 
-  test("test logout function fails", async () => {
-    jest.spyOn(window, "alert").mockImplementation(() => {});
-
+  test("test logout function", async () => {
     await act(async () => {
       const logoutButton = screen.getByTestId("logout-button");
       await userEvent.click(logoutButton);
     });
+    expect(location.pathname).toEqual("/");
+  });
 
-    expect(window.alert).toHaveBeenCalledWith("failed!");
+  test("test login with IDM function", async () => {
+    await act(async () => {
+      const loginButton = screen.getByTestId("login-idm-button");
+      await userEvent.click(loginButton);
+    });
+    expect(screen.getByTestId("testdiv")).toHaveTextContent("User Test");
   });
+});
 
-  test("test logout function", async () => {
-    // stash alert and define temporary mock for use by jest-dom
-    const actualAlert = window.alert;
-    window.alert = () => {};
+describe("Test UserProvider with production path", () => {
```

Comment:
Just added in helper functions here and moved things around for cleanliness. This whole file has a lot of things that got moved around a bit but were actually there before. Sadly we lack the diffing technology in our time to show that.